src/TransitionState.php

```diff
@@ -6,5 +6,5 @@
 
 interface TransitionState
 {
-    public function __toString();
+    public function getValue(): string;
```

Comment:
Now, this is out of scope, just wanted to put my thoughts out there of something that I noticed:

In our own domain we can always assume that only an enum will implement this `TransitionState` interface. I get that this is done so other's can implement their own way and it's not bound to an enum, but is it worth it considering that we have 0 other's using this package? 😀 

Nothing to change for now, but I was just wondering personally 😄